### ISSUE TITLE
Wrong information in .xmod

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#308]
With the following code, the module file created cannot be imported. The `FbasicType` `C7febb740a310` has wrong information for the length as instead of having `is_assumed_shape=true`, it has the local variable `nchars`

```fortran
module mod_issue308
contains
function c_str_to_string(s) result(string)
  character(len=:), allocatable :: string
  integer :: nchars
  allocate(character(len=nchars) :: string)
end function
end module mod_issue308
```

```xml
<OmniFortranModule version="1.0">
  <name>mod_issue308</name>
  <depends>
  </depends>
  <typeTable>
    <FfunctionType type="F7febb7408160" result_name="string" return_type="C7febb7408520">
      <params>
        <name type="Freal">s</name>
      </params>
    </FfunctionType>
    <FbasicType type="C7febb7408f20" is_allocatable="true" ref="Fcharacter">
      <len is_assumed_shape="true">
      </len>
    </FbasicType>
    <FbasicType type="C7febb7408520" is_allocatable="true" ref="C7febb7408f20"/>
    <FbasicType type="C7febb7409120" is_allocatable="true" ref="C7febb7408f20"/>
    <FbasicType type="C7febb740a310" ref="Fcharacter">
      <len>
        <Var scope="local">nchars</Var>
      </len>
    </FbasicType>
  </typeTable>
  <identifiers>
    <id type="F7febb7408160" sclass="ffunc" original_name="c_str_to_string" declared_in="mod_issue308">
      <name>c_str_to_string</name>
    </id>
  </identifiers>
  <declarations>
  </declarations>
  <interfaceDecls>
  </interfaceDecls>
  <aux_info>
  </aux_info>
</OmniFortranModule>
```

```fortran
program issue308
  use mod_issue308
end program issue308
```

Then when we try to use this module, an error occurs: 
```bash
"../test/testdata/issue308.f90", line 2: failed to import module 'mod_issue308'
```

**Comment by clementval** [comment]
I think the `FbasicType` should not be present in the `.xmod` file 

```xml
<FbasicType type="C7febb740a310" ref="Fcharacter">
  <len>
    <Var scope="local">nchars</Var>
  </len>
</FbasicType>
```

**Comment by h-murai** [comment]
@clementval Is this issue fixed along with #533 ? If so, please close this.

**Comment by clementval** [comment]
This is fixed
